src/PrintBuffer.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "Defs.hpp"
+
+#include <string>
+
+namespace yb {
+
+class PrintBuffer final {
+
+    using Container = std::string;
+    using Iterator = std::string::const_iterator;
+    Container buffer_;
+    Iterator position_;
+    const size_t defaultBufferSize_ = 1024;
+
+public:
+
+    PrintBuffer() {
+        buffer_.reserve(defaultBufferSize_);
+        position_ = buffer_.end();
+    }
+
+    explicit PrintBuffer(size_t size) {
+        buffer_.reserve(size);
+        position_ = buffer_.end();
+    }
+
+    PrintBuffer &operator =(std::string &&s) {
```

Comment:
are you sure we should use `operator=` here? it is quite close to move assignment operator and I guess it wasn't your intention?